src/ImpactX.cpp

```diff
@@ -199,6 +212,9 @@ namespace impactx
 
                 }
 
+                // inputs: unused parameters (e.g. typos) check after step 1 has finished
+                if (!early_params_checked) { early_param_check(); early_params_checked = true;}
```

Comment:
See API change:
```suggestion
                if (!early_params_checked) { early_params_checked = early_param_check(); }
```

(we could have also passed by reference, but I like being explicit a bit more.)